### PR TITLE
Add subtraction upload endpoints

### PR DESCRIPTION
**Pull request by tianshengsui**
[ch375]

### CODE REVIEW

**Comment by igboyes** [comment]
A file needs to be uploaded in this request. The `name` needs to be in the URL query, not a JSON body. See how this is done in `virtool.uploads`.

Please find a way to reuse code from the modules in `virtool.uploads`. Discuss with @jakeale so you stay in sync on this and any changes necessary to extend this code for subtraction uploads.

**Comment by tianshengsui** [comment]
@igboyes Can I assume the upload file will be stored at `<data_path>/subtractions/:id/<file_name>`?

**Comment by igboyes** [comment]
> @igboyes Can I assume the upload file will be stored at `<data_path>/subtractions/:id/<file_name>`?

Yes. And that there is only that strict set of allowed files names defined in the story.

File: virtool/subtractions/api.py [hunk 1]
```diff
@@ -173,6 +185,69 @@ async def create(req):
...
+
+    file = {
+        "name": file_name,
+        "size": size,
+        "type": file_type
+    }
+
+    await db.subtraction.find_one_and_update({"_id": subtraction_id}, {
+        "$push": {
+            "files": file
```
Comment (igboyes) [inline virtool/subtractions/api.py:240]: Store file IDs in this array. They will be used to join the SQL records to the subtraction when it is requested.

**Review by igboyes — CHANGES_REQUESTED** [review]
Very good. One change required.

File: virtool/subtractions/api.py (hunk 1 above)
Comment (tianshengsui) [reply virtool/subtractions/api.py:240]: Only file IDs?

**Review by igboyes — APPROVED** [review]

**Pull request merged**